src/3d/symbols/qgspointcloud3dsymbol.cpp

```diff
@@ -15,46 +15,213 @@
 
 #include "qgspointcloud3dsymbol.h"
 
+#include "qgscolorramptexture.h"
 
-QgsPointCloud3DSymbol::QgsPointCloud3DSymbol()
+#include <Qt3DRender/QMaterial>
+#include <Qt3DRender/QParameter>
+#include <Qt3DRender/QTexture>
+
+// QgsPointCloud3DSymbol
+
+QgsPointCloud3DSymbol::QgsPointCloud3DSymbol( QgsPointCloud3DSymbol::RenderingStyle style )
   : QgsAbstract3DSymbol()
+  , mRenderingStyle( style )
 {
-
 }
 
 QgsPointCloud3DSymbol::~QgsPointCloud3DSymbol() {  }
 
-QgsAbstract3DSymbol *QgsPointCloud3DSymbol::clone() const
+void QgsPointCloud3DSymbol::setPointSize( float size )
+{
+  mPointSize = size;
+}
+
+// QgsSingleColorPointCloud3DSymbol
+
+QgsSingleColorPointCloud3DSymbol::QgsSingleColorPointCloud3DSymbol()
+  : QgsPointCloud3DSymbol( QgsPointCloud3DSymbol::RenderingStyle::SingleColor )
+{
+
+}
+
+QgsAbstract3DSymbol *QgsSingleColorPointCloud3DSymbol::clone() const
 {
-  QgsPointCloud3DSymbol *result = new QgsPointCloud3DSymbol;
-  result->mEnabled = mEnabled;
+  QgsSingleColorPointCloud3DSymbol *result = new QgsSingleColorPointCloud3DSymbol;
   result->mPointSize = mPointSize;
+  result->mSingleColor = mSingleColor;
   copyBaseSettings( result );
   return result;
 }
 
-void QgsPointCloud3DSymbol::writeXml( QDomElement &elem, const QgsReadWriteContext &context ) const
+void QgsSingleColorPointCloud3DSymbol::writeXml( QDomElement &elem, const QgsReadWriteContext &context ) const
 {
   Q_UNUSED( context )
 
-  elem.setAttribute( QStringLiteral( "enabled" ), mEnabled );
   elem.setAttribute( QStringLiteral( "point-size" ), mPointSize );
+  elem.setAttribute( QStringLiteral( "rendering-style" ), mRenderingStyle );
+  elem.setAttribute( QStringLiteral( "single-color-red" ), mSingleColor.redF() );
+  elem.setAttribute( QStringLiteral( "single-color-green" ), mSingleColor.greenF() );
+  elem.setAttribute( QStringLiteral( "single-color-blue" ), mSingleColor.blueF() );
```

Comment:
please use QgsSymbolLayerUtils::encodeColor() and decodeColor() when loading and saving colors for consistency